scripts/gas_diff_stats.py

```diff
@@ -99,42 +104,62 @@ def try_parse(line):
 
 def semantictest_statistics():
     """Prints the tabulated statistics that can be pasted in github."""
-    def try_parse_git_diff(fname):
-        try:
-            diff_output = subprocess.check_output(
-                "git diff --unified=0 origin/develop HEAD " + fname,
-                shell=True,
-                universal_newlines=True
-            ).splitlines()
-            if diff_output:
-                return collect_statistics(diff_output)
-        except subprocess.CalledProcessError as e:
-            print("Error in the git diff:")
-            print(e.output)
-        return None
+    def parse_git_diff(fname):
+        diff_output = subprocess.check_output(
+            ["git", "diff", "--unified=0", "origin/develop", "HEAD", fname],
+            universal_newlines=True
+        ).splitlines()
+        if len(diff_output) == 0:
+            return None
+        return collect_statistics(diff_output)
+
     def stat(old, new):
-        return ((new - old) / old) * 100  if old else 0
+        if old == 0:
+            return ''
+        percentage = (new - old) / old * 100
+        prefix = (
+            # Distinguish actual zero from very small differences
+            '+' if round(percentage) == 0 and percentage > 0 else
+            '-' if round(percentage) == 0 and percentage < 0 else
+            ''
+        )
+        return f'{prefix}{round(percentage)}%'
 
     table = []
 
-    for path in Path("test/libsolidity/semanticTests").rglob("*.sol"):
+    if not SEMANTIC_TEST_DIR.is_dir():
+        sys.exit(f"Semantic tests not found. '{SEMANTIC_TEST_DIR.absolute()}' is missing or not a directory.")
+
+    for path in SEMANTIC_TEST_DIR.rglob("*.sol"):
         fname = path.as_posix()
-        parsed = try_parse_git_diff(fname)
+        parsed = parse_git_diff(fname)
         if parsed is None:
             continue
         ir_optimized = stat(parsed[0], parsed[3])
         legacy_optimized = stat(parsed[1], parsed[4])
         legacy = stat(parsed[2], parsed[5])
-        fname = fname.split('/', 3)[-1]
-        table += [map(str, [fname, ir_optimized, legacy_optimized, legacy])]
+        fname = f"`{fname.split('/', 3)[-1]}`"
+        table += [[fname, ir_optimized, legacy_optimized, legacy]]
 
     if table:
         print("<details><summary>Click for a table of gas differences</summary>\n")
-        table_header = ["File name", "IR-optimized (%)", "Legacy-Optimized (%)", "Legacy (%)"]
-        print(tabulate(table, headers=table_header, tablefmt="github"))
+        table_header = ["File name", "IR optimized", "Legacy optimized", "Legacy"]
+        print(tabulate(sorted(table), headers=table_header, tablefmt="github"))
```

Comment:
It might also be useful to order the results by magnitude of the change.

There are three numbers so we'd have to sum or average their absolute values for it to make sense, but it would probably make reviewing results easier, especially when the table is long.